### DISCUSSION TITLE
Suggestion: add a step "security scan of the model"

### DISCUSSION THREAD

**Discussion by darth2602** [TheLastBen/fast-stable-diffusion#525] [Ideas]
Hello, I have a suggestion of important function to add:

I saw on youtube that ckpt files can be infected by "pickels" (malware ?) 

On pc you can scan them with "pickle_inspector.py" and "pickle_scan.py" but you cannot do anything if you use automatic1111 from google colab... 

For google colab users can you add to the script an automatic 
automatic scan, for example when loading the ckpt template   
at the "Start stable-diffusion" step of fast_stable_diffusion_AUTOMATIC1111.ipynb
with the options to check in the cell :  

- scan the model before loading it
- delete the infected model (and load model.ckpt instead)

it could also be an optional cell added between 

Before
"Installing Requirements
"Start stable-diffusion

After
"Installing Requirements
"Scanning models (Detecting models infected by Pickels)"   
"Start stable-diffusion"

and why not add a tab scanner in web ui (extension) to scan the ckpt of our choice with options like : 

- scan selected templates (templates selectable from the list of all installed templates)

- scan all models 
- delete infected models

thanks in advance ...
   
the youtube video : 
https://www.youtube.com/watch?v=YFHFl9NEx8A

it's just a suggestion; maybe the threat is not so important and the video is alarmist, maybe also the 
infected ckpt have no effect with google colab (only on pc) 

anyway i think we should still be able to detect and remove them from fast_stable_diffusion_AUTOMATIC1111.ipynb installed and launched on google colab...

Sincerely

**Comment by TheLastBen** [comment]
most models now come in safetensors (safe), if you don't download a ckpt from unknown sources you're safe
